File: measurement-generator/src/main/java/pl/polsl/hdised/measurementgenerator/measurement/MeasurementController.java
package pl.polsl.hdised.measurementgenerator.measurement;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


@RestController
@RequestMapping("api/v1/measurements")
public class MeasurementController {

    private final String TOPIC = "topic";
    private final String STREAM_TOPIC = "streamTopic";

    private final MeasurementService measurementService;

    public MeasurementController(MeasurementService measurementService) {
        this.measurementService = measurementService;
    }

    @PostMapping("/stop-producing")
    public String stopProducing() {
        this.measurementService.stopProducing();
        return "Producing stopped";
    }

    @PostMapping("/start-producing-to-database")
    public String startProducingToDatabase() {
        this.measurementService.startProducing(TOPIC);
        return "Producing started";
    }

    @PostMapping("/start-stream-producing")
    public String startStreamProducing() {
        this.measurementService.startProducing(STREAM_TOPIC);
        return "Producing started";
    }

}

File: golang-measurement-generator/measurementController.go
package main

import (
	"net/http"

	"github.com/gorilla/mux"
)

type MeasurementController struct {
	MeasurementService *MeasurementService
}

func NewMeasurementController(service *MeasurementService) *MeasurementController {
	return &MeasurementController{
		MeasurementService: service,
	}
}

func (controller *MeasurementController) StartRouter() {
	router := mux.NewRouter()
	router.HandleFunc("/api/v1/measurements/stop-producing", controller.StopProducing).Methods("POST")
	router.HandleFunc("/api/v1/measurements/start-producing-to-database", controller.StartProducingToDatabase).Methods("POST")
	router.HandleFunc("/api/v1/measurements/start-stream-producing", controller.StartStreamProducing).Methods("POST")
	http.ListenAndServe(":8085", router)
}

func (controller *MeasurementController) StopProducing(w http.ResponseWriter, r *http.Request) {
	controller.MeasurementService.StopProducing()
	w.WriteHeader(http.StatusOK)
	w.Write([]byte("Producing stopped"))
}

func (controller *MeasurementController) StartProducingToDatabase(w http.ResponseWriter, r *http.Request) {
	controller.MeasurementService.StartProducing("topic")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte("Producing started"))
}

func (controller *MeasurementController) StartStreamProducing(w http.ResponseWriter, r *http.Request) {
	controller.MeasurementService.StartProducing("streamTopic")
	w.WriteHeader(http.StatusOK)
	w.Write([]byte("Producing started"))
}
